### ISSUE TITLE
docker::run on Windows uses "\" for multiline command instead of "`"

### ISSUE THREAD

**Issue by witsec** [puppetlabs/puppetlabs-docker#314]
## What you expected to happen?
I was trying out the new Docker 2.0.0 module on Windows Server 2016, see how far I could get. I ended up installing Docker using a Docker EE Chocolatey package rather than make use of `class { 'docker': }`, as that was a pain. But this issue is not about that.

I'm trying to run a container, using the following code in my manifest:

```puppet
docker::run { 'portainer':
  image           => 'portainer/portainer',
  detach          => true,
  ports           => ['9000'],
  expose          => ['9000'],
  username        => 'administrator',
  restart_service => true,
  restart         => 'always',
}
```

## What happened?
I get an error message:

```
Error: 'docker run -d -u 'administrator' \
--net bridge \
-m 0b \
--detach=true \
--expose=9000 \
 \
-p 9000 \
 \
 --name portainer --cidfile=c:/Windows/Temp/docker-portainer.cid --restart="always" portainer/portainer ' returned 1 instead of one of [0]
Error: /Stage[main]/Profile::Windows_docker_portainer/Docker::Run[portainer]/Exec[run portainer with docker]/returns: change from 'notrun' to ['0'] failed: 'docker run -d -u 'administrator' \
--net bridge \
-m 0b \
--detach=true \
--expose=9000 \
 \
-p 9000 \
 \
 --name portainer --cidfile=c:/Windows/Temp/docker-portainer.cid --restart="always" portainer/portainer ' returned 1 instead of one of [0]
```

It looks like it's caused by the added blackslashes. Powershell (=exec_provider) doesn't use that, instead it uses "`" for multiline commands.

Example:
```
docker `
version
```

After doing some more research, I found the issue. In the function `docker_run_flags.rb` there's this line:

```
params_join_char = Facter.value(:osfamily).casecmp('windows').zero? ? ' ' : " \\\n"
```

This is supposed to return a space instead of backslash, but that doesn't work if you're using a Puppetmaster. Since a function runs on the master, which is Linux, this returns "RedHat" (for me), and so casecmp('windows') does not return 0, resulting in the commands featuring a backslash and break line.

## How to reproduce it?
Run my Puppet code on Windows and sit back. Also make use of a Puppetmaster.

## Anything else we need to know?
No.

## Versions:
```
$ puppet --version
5.5.3

$ docker version
Client:
 Version:      17.06.2-ee-14
 API version:  1.30
 Go version:   go1.8.7
 Git commit:   6345dd7
 Built:        Thu Jun 21 18:16:45 2018
 OS/Arch:      windows/amd64

Server:
 Engine:
  Version:      17.06.2-ee-14
  API version:  1.30 (minimum version 1.24)
  Go version:   go1.8.7
  Git commit:   6345dd7
  Built:        Thu Jun 21 18:28:51 2018
  OS/Arch:      windows/amd64
  Experimental: false

$ facter os
{
  architecture => "x64",
  family => "windows",
  hardware => "x86_64",
  name => "windows",
  release => {
    full => "2016",
    major => "2016"
  },
  windows => {
    system32 => "C:\Windows\system32"
  }
}

$ puppet module list
C:/ProgramData/PuppetLabs/code/environments/production/modules (no modules installed)
C:/ProgramData/PuppetLabs/code/modules (no modules installed)
```

## Logs:
If I do a `puppet agent -dt`, so with added debug, the logs show this:
```
Debug: STDERR: Error: No such object: portainer
Debug: /Stage[main]/Profile::Windows_docker_portainer/Docker::Run[portainer]/Exec[run portainer with docker]/unless: []
Debug: Waited 0.0 total seconds.
Debug: STDERR: At line:2 char:3
+ --net bridge \
+   ~
Missing expression after unary operator '--'.

At line:2 char:3
+ --net bridge \
+   ~~~
Unexpected token 'net' in expression or statement.
```
...and so on.

**Comment by davejrt** [comment]
@witsec thanks for all the info. I'll raise a bug internally and someone will back to you when it's being worked.

**Comment by davejrt** [comment]
Fixed in #317
